pkg/controller/registry/reconciler/grpc.go

```diff
@@ -199,11 +201,24 @@ func (c *GrpcRegistryReconciler) ensureUpdatePod(source grpcCatalogSourceDecorat
 
 			// Update the update pod to promote it to serving pod
 			_, err := c.OpClient.KubernetesInterface().CoreV1().Pods(source.GetNamespace()).Update(context.TODO(), updatePod, metav1.UpdateOptions{})
-			if err != nil {
-				return errors.Wrapf(err, "error creating new pod: %s", source.Pod().GetName())
+			if err == nil {
+				break
 			}
+			if !(apierrors.IsConflict(err) && err.Error() == registry.OptimisticLockErrorMsg) {
+				return errors.Wrapf(err, "error updating catalog source pod labels: %s", source.Pod().GetName())
```

Comment:
Why not use fmt.errorf()?

Reply:
I'm guessing that was for the stacktrace, we can switch to errorf